src/zjs_board.c

```diff
@@ -0,0 +1,188 @@
+//    exposing distinct functionality; unclear how to understand this so far
+static const zjs_pin_t pin_data[] = {
+    {"IO0",  "0",  NULL,   0, 255,   9, 255},
+    {"IO1",  "1",  NULL,   0, 255,   8, 255},
+    {"IO2",  "2",  NULL,   0,  18, 255, 255},
+    {"IO3",  "3",  "PWM0", 0,  17,  10,   0},
+    {"IO4",  "4",  NULL,   0,  19, 255, 255},
+    {"IO5",  "5",  "PWM1", 0,  15,  11,   1},
+    {"IO6",  "6",  "PWM2", 0, 255,  12,   2},
+    {"IO7",  "7",  NULL,   0,  20, 255, 255},
+    {"IO8",  "8",  NULL,   0,  16, 255, 255},
+    {"IO9",  "9",  "PWM3", 0, 255,  13,   3},
+    {"IO10", "10", NULL,   0,  11,   0, 255},
+    {"IO11", "11", NULL,   0,  10,   3, 255},
+    {"IO12", "12", NULL,   0,   9,   1, 255},
+    {"IO13", "13", "LED2", 0,   8,   2, 255},
+    {"AD0",  "A0", NULL,   0, 255,  10, 255},
+    {"AD1",  "A1", NULL,   0, 255,  11, 255},
+    {"AD2",  "A2", NULL,   0, 255,  12, 255},
+    {"AD3",  "A3", NULL,   0, 255,  13, 255},
+    {"AD4",  "A4", NULL,   0, 255,  14, 255},
+    {"AD5",  "A5", NULL,   0, 255,   9, 255},
+    {"LED0", NULL, NULL,   0,  26, 255, 255},
+    {"LED1", NULL, NULL,   0,  12, 255, 255}
+};
+#elif CONFIG_BOARD_FRDM_K64F
+enum gpio_ports {
+    PTA, PTB, PTC, PTD, PTE
+};
+
+static const zjs_pin_t pin_data[] = {
+    {"D0",    "0",  NULL, PTC, 16, 255},
+    {"D1",    "1",  NULL, PTC, 17, 255},
+    {"D2",    "2",  NULL, PTB,  9, 255},
+    {"D3",    "3",  NULL, PTA,  1, 255},
+    {"D4",    "4",  NULL, PTB, 23, 255},
+    {"D5",    "5",  NULL, PTA,  2, 255},
+    {"D6",    "6",  NULL, PTC,  2, 255},
+    {"D7",    "7",  NULL, PTC,  3, 255},
+    // currently not working on rev E3; used to work as input/output
+    {"D8",    "8",  NULL, PTC, 12, 255},  // PTA0 for Rev <= D
+    {"D9",    "9",  NULL, PTC,  4, 255},
+    {"D10",  "10",  NULL, PTD,  0, 255},
+    {"D11",  "11",  NULL, PTD,  2, 255},
+    {"D12",  "12",  NULL, PTD,  3, 255},
+    {"D13",  "13",  NULL, PTD,  1, 255},
+    {"D14",  "14",  NULL, PTE, 25, 255},
+    {"D15",  "15",  NULL, PTE, 24, 255},
+    {"A0",   NULL,  NULL, PTB,  2, 255},
+    {"A1",   NULL,  NULL, PTB,  3, 255},
+    {"A2",   NULL,  NULL, PTB, 10, 255},
+    {"A3",   NULL,  NULL, PTB, 11, 255},
+    {"A4",   NULL,  NULL, PTC, 11, 255},
+    {"A5",   NULL,  NULL, PTC, 10, 255},
+    {"LEDR", NULL,  NULL, PTB, 22, 255},
+    {"LEDG", NULL,  NULL, PTE, 26, 255},
+    {"LEDB", NULL,  NULL, PTB, 21, 255},
+    {"SW2",  NULL,  NULL, PTC,  6, 255},
+    {"SW3",  NULL,  NULL, PTA,  4, 255},
+    {"PWM0", NULL,  NULL, PTA,  1,   0},
+    {"PWM1", NULL,  NULL, PTA,  2,   2},
+    {"PWM2", NULL,  NULL, PTC,  2,   3},
+    {"PWM3", NULL,  NULL, PTC,  3,   3},
+    {"PWM4", NULL,  NULL, PTC, 12,   4},
+    {"PWM5", NULL,  NULL, PTC,  4,   5},
+    {"PWM6", NULL,  NULL, PTD,  0,   6},
+    {"PWM7", NULL,  NULL, PTD,  2,   7},
+    {"PWM8", NULL,  NULL, PTD,  3,   8},
+    {"PWM9", NULL,  NULL, PTD,  1,   9}
+    // TODO: More pins at https://developer.mbed.org/platforms/FRDM-K64F/
+};
```

Comment:
That's true, but let's cross that bridge when we come to it. For now I want to move things out of the pins modules because those will be disappearing as require() modules at least (Not yet because pwm and aio still use them.) The way this board api works now, they will be implicitly defined given how you build with BOARD.